### PR TITLE
feat: add semantic PR check to js-libp2p repos

### LINKED ISSUE #81: [General] Add ci check to ensure PR titles follows semantic release rules
### Summary

Add github workflow to every js-libp2p* repo to check if PR title follows semantic release rules.
This action works great for us at chainsafe: https://github.com/amannn/action-semantic-pull-request

### Why do you need this?

To prevent accidentally merging community PR with invalid title which doesn't show in changelog.

### What else do we need to know?

**DRI:** myself

### PR DESCRIPTION
**Pull request by mpetrunic**
### Summary

Resolves #81

### What else do we need to know?

**DRI:** myself

### Reviewer's Checklist

- [ ] It is clear where the request is coming from (if unsure, ask)
- [ ] All the automated checks passed
- [ ] The YAML changes reflect the summary of the request
- [ ] The Terraform plan posted as a comment reflects the summary of the request

### CODE REVIEW

**Comment by mpetrunic** [comment]
cc @achingbrain @wemeetagain

File: .github/workflows/action-semantic-pull-request.yml
Comment (galargh) [inline .github/workflows/action-semantic-pull-request.yml:1]: Let's move this file here: https://github.com/mpetrunic/github-mgmt/tree/feat/js-libp2p-semantic-prs/files/.github/workflows

The difference is that `.github/workflows` is a directory for workflows that are to be executed in this repo, while `files/.github/workflows` is intended for the workflow files that we want to "copy" to other repos.

File: .github/workflows/action-semantic-pull-request.yml
Comment (galargh) [inline .github/workflows/action-semantic-pull-request.yml:27]: Newline at the EOF
```suggestion
          requre_scope: false

```

**Review by galargh — COMMENTED** [review]
Thanks for creating the PR 😄 Once we move the file, it'll be good to go from the GitHub Management point of view. I'll let others review before the merge though.

File: .github/workflows/action-semantic-pull-request.yml
Comment (achingbrain) [inline .github/workflows/action-semantic-pull-request.yml:15]: The general policy appears to be to depend on specific commits of third-party actions rather than version numbers/ranges.

Comment (galargh) [reply .github/workflows/action-semantic-pull-request.yml:15]: Good catch! Yeah, let's change it to
```suggestion
      - uses: amannn/action-semantic-pull-request@01d5fd8a8ebb9aafe902c40c53f0f4744f7381eb # v5.0.2
```

**Comment by achingbrain** [comment]
I've not used this plugin before, does it pull the list of allowed scopes from the [config](https://github.com/ipfs/aegir/blob/master/src/check-project/semantic-release-config.js#L11-L69) or does it need configuring in the action yaml?

File: files/.github/workflows/semantic-pull-request.yml
Comment (galargh) [inline files/.github/workflows/semantic-pull-request.yml:15]: When something's off or we want to upgrade, it comes in quite handy to leave the version string in a comment:

```suggestion
      - uses: amannn/action-semantic-pull-request@01d5fd8a8ebb9aafe902c40c53f0f4744f7381eb # v5.0.2
```

**Comment by mpetrunic** [comment]
> I've not used this plugin before, does it pull the list of allowed scopes from the [project config](https://github.com/ipfs/aegir/blob/master/src/check-project/semantic-release-config.js#L11-L69) or does it need configuring in the action yaml?

https://github.com/libp2p/github-mgmt/pull/93/files#diff-e23c8debbc471dd71fbc4c6a03ce8197bc673f0470a994c63be03769c5096917R20

**Comment by p-shahi** [comment]
@mpetrunic is all that's remaining for this a review? @thomaseizinger added pretty much the same thing in rust-libp2p so I think that's proof that this works.

We'd want to enable this in go-libp2p too per: https://github.com/libp2p/go-libp2p/issues/1931 cc @MarcoPolo

**Review by p-shahi — APPROVED** [review]

**Comment by mpetrunic** [comment]
> @mpetrunic is all that's remaining for this a review? @thomaseizinger added pretty much the same thing in rust-libp2p so I think that's proof that this works. I approved but feel free to wait for @achingbrain
> 
> We'd want to enable this in go-libp2p too per: [libp2p/go-libp2p#1931](https://github.com/libp2p/go-libp2p/issues/1931) cc @MarcoPolo

Yup, think we are all waiting for approves here^^

File: files/.github/workflows/semantic-pull-request.yml
Comment (laurentsenta) [inline files/.github/workflows/semantic-pull-request.yml:27]:
```suggestion
          requireScope: false
```

Missing `i` + [casing](https://github.com/amannn/action-semantic-pull-request/blob/b6bca70dcd3e56e896605356ce09b76f7e1e0d39/README.md?plain=1#L71)?

Comment (mpetrunic) [reply files/.github/workflows/semantic-pull-request.yml]: tnx

**Comment by thomaseizinger** [comment]
It works quite well for us!

Something I've also added is a lint for the length of the PR title to not exceed 72 chars. This might be interesting to include here as well.

File: files/.github/workflows/semantic-pull-request.yml
Comment (thomaseizinger) [inline files/.github/workflows/semantic-pull-request.yml:26]: We also allow `refactor` and `ci`. Something to consider :)

**Review by thomaseizinger — COMMENTED** [review]
cc @mxinden We should probably switch to this being managed through here as well, assuming we can agree on the set of scopes :)

**Review by achingbrain — APPROVED** [review]
LGTM

File: files/.github/workflows/semantic-pull-request.yml
Comment (p-shahi) [reply files/.github/workflows/semantic-pull-request.yml:26]: I added those two here.

**Comment by p-shahi** [comment]
> I'll let others review before the merge though.

@galargh I think you've already reviewed but I don't want to merge this without your approval. PTAL

**Review by galargh — APPROVED** [review]

**Pull request merged**
